Backend/routes/admin.js

```diff
@@ -0,0 +1,310 @@
+const express = require('express');
+const router = express.Router();
+const { checkSchema, validationResult } = require('express-validator');
+const validacion = require('../utils/validacionesRecetas');
+const funcionesToken = require('../utils/token');
+
+//#region Controllers
+
+const recetaController = require('../controllers/recetaController');
+
+//#endregion
+
+//#region Rutas
+
+router.post('/agregarReceta', checkSchema(validacion), (req, res) => {
+    if (
+        typeof req.body.titulo == 'undefined' ||
+        typeof req.body.email == 'undefined' ||
+        typeof req.body.descripcion == 'undefined' ||
+        typeof req.body.imagen == 'undefined' ||
+        typeof req.body.ingredientes == 'undefined' ||
+        typeof req.body.pasos == 'undefined'
+    ) {
+        res.status(400).json('Error: Campos incompletos');
+        return;
+    }
+    const resValidaciones = validationResult(req).array();
+    if (resValidaciones.length > 0) {
+        console.log(resValidaciones);
+        res.send({
+            success: false,
+            message: 'Campos inválidos',
+            content: resValidaciones,
+        });
+        return;
+    }
+
+    const categorias = req.body.categorias ? `'${JSON.stringify(req.body.categorias)}'` : null;
+    const tiempoCoccion = req.body.tiempoCoccion ? `'${req.body.tiempoCoccion}'` : null;
+    const dificultad = req.body.dificultad ? `'${req.body.dificultad}'` : null;
+
+    db.query(
+        `CALL sp_crearReceta('${req.body.titulo}', '${req.body.email}', ${tiempoCoccion}, ${dificultad}, '${req.body.descripcion}',
```

Comment:
Cuando corregí el procedimiento le cambié el orden a los parámetros, ahora el mail va primero. Te paso todos por si acaso:
(IN emailCreador VARCHAR(250), 
IN tituloR VARCHAR(250) , 
IN tiempoCoccionR VARCHAR(20), 
IN dificultadR VARCHAR(12), 
IN descripcionR TEXT, 
IN imagenR TEXT, 
IN ingredientes JSON , 
IN pasos JSON, 
IN categorias JSON)

Por otra parte, las categorías también deberían pasar por stringify, aunque como no las vamos a implementar no es necesario, podés dejarlo en null.

La imagen va a ser la url, por eso es un TEXT ahora.

Reply:
Este endpoint aun falta por actualizar